sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -859,11 +859,11 @@ case class FromUnixTime(sec: Expression, format: Expression, timeZoneId: Option[
     } else {
       val zid = ctx.addReferenceObj("zoneId", zoneId, classOf[ZoneId].getName)
       val locale = ctx.addReferenceObj("locale", Locale.US)
-      val tf = TimestampFormatter.getClass.getName.stripSuffix("$")
+      val tf = classOf[Iso8601TimestampFormatter].getName
       nullSafeCodeGen(ctx, ev, (seconds, f) => {
         s"""
         try {
-          ${ev.value} = UTF8String.fromString($tf.apply($f.toString(), $zid, $locale).
```

Comment:
is it caused by some Scala-Java interoperation problems?

Reply:
Hm, yeah I'm a little confused too. There is a `TimestampFormatter.apply` method defined on the companion object. `TimestampFormatter` is a trait (interface). You're allowed static methods in interfaces in Java 8. I wonder indeed if it's a _Scala_ issue somehow with how the method is generated, like it's not quite tagged as static.

Anyway I agree with the fix and we can leave a todo, maybe, about what we know of the issue.

Reply:
Scala seems to fix this issue bacause `TimestampFormatter.apply` works in `scala-shell`. `Janino` seems to be behind.

Reply:
Scala seems to fix this issue bacause `TimestampFormatter.apply` works in `scala-shell`. `Janino` seems to be behind.

Reply:
This is Java code that Janino's compiling. Still, `TimestampFormatter.apply()` seemed legal in some test Java class I tried locally. OK yeah, I mean it might be a Janino thing not Scala, right. There is a slightly newer janino (3.0.12 vs 3.0.11) but I don't see any relevant fixes.